From a0a2c86e133e0f0453dfd545ac8a8125b2a8d995 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Wed, 4 Oct 2023 10:03:09 +0200
Subject: [PATCH 0514/1695] dts: stm32: declare the reboot modes in
 stm32mp251.dtsi

reboot mode is a consumer of nvmem cell

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4fda2d980565cd9294137729d3196130e946a57d dts: stm32: declare the reboot modes in stm32mp251.dtsi]
Change-Id: I51772e94cb50b26bbc131f02fe020ae820d2e284
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/328931
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
(cherry picked from commit 72d2454456cefab7776163ce5ca74f65a4cc8ae6)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351779
---
 core/arch/arm/dts/stm32mp251.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index d55119da2..b9ebf936c 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -775,6 +775,20 @@
 					reg = <0x180 0x4>;
 				};
 			};
+
+			reboot_mode: reboot-mode {
+				compatible = "nvmem-reboot-mode";
+				nvmem-cells = <&boot_mode>;
+				nvmem-cell-names = "reboot-mode";
+				mode-normal = <0x00>;
+				mode-fastboot = <0x01>;
+				mode-recovery = <0x02>;
+				mode-stm32cubeprogrammer = <0x03>;
+				mode-ums_mmc0 = <0x10>;
+				mode-ums_mmc1 = <0x11>;
+				mode-ums_mmc2 = <0x12>;
+				mode-romcode_serial = <0xff>;
+			};
 		};
 
 		pinctrl_z: pinctrl-z@46200000 {
-- 
2.47.2

